scripts/library.sh

```diff
@@ -513,6 +513,63 @@ function add_trap {
   done
 }
 
+# Update go deps.
+# Parameters (parsed as flags):
+#   "--upgrade", bool, do upgrade.
+#   "--release <version>" used with upgrade. The release version to upgrade
+#                         Knative components. ex: --release v0.18. Defaults to
+#                         "master".
+function go_update_deps() {
+  cd "${REPO_ROOT_DIR}" || return 1
+
+  export GO111MODULE=on
+  export GOFLAGS=""
+
+  echo "=== Update Deps for Golang"
+
+  local UPGRADE=0
+  local VERSION="master"
+  while [[ $# -ne 0 ]]; do
+    parameter=$1
+    case ${parameter} in
+      --upgrade) UPGRADE=1 ;;
+      --release) shift; VERSION="$1" ;;
+      *) abort "unknown option ${parameter}" ;;
+    esac
+    shift
+  done
+
+  if (( UPGRADE )); then
+    echo "--- Upgrading to ${VERSION}"
+    FLOATING_DEPS=( $(run_go_tool knative.dev/test-infra/buoy buoy float ${REPO_ROOT_DIR}/go.mod --release ${VERSION} --domain knative.dev) )
```

Comment:
we need the ability to add non-knative things here.

Reply:
I don't mean to buoy, but to FLOATING_DEPS

Reply:
ok contour will have to do this:

(update_deps.sh):

```
source $(dirname "$0")/../scripts/library.sh

CONTOUR_VERSION="v1.9.0" # This is for controlling which version of contour we want to use.

FLOATING_DEPS=(
  "github.com/projectcontour/contour@${CONTOUR_VERSION}"
)

go_update_deps "$@"
```